build/setup.sh

```diff
@@ -23,9 +23,27 @@ else
 fi
 COMMIT_DATE="$(git log -n1 --pretty='format:%cd' --date=format:'%Y%m%dT%H%M%S')"
 
+# Arrange to write the docker container ID to a temp file, then run
+# the container detached and immediately attach it (without stdin) so
+# we can catch CTRL-C in this script and kill the container if so.
+
+DOCKER_CONTAINER_ID_FILE=$(mktemp)
+rm $DOCKER_CONTAINER_ID_FILE
```

Comment:
I understand that this line is to remove file if it existed previously, but the only way for it to exist is if it was created with the docker run, that's run with sudo, so I would use sudo in this rm command also.

Reply:
It's created by that `mktemp` on line 30, and therefore owned by the user.  Maybe `mktemp --dry-run` would be better.  I'll look at it a bit more.